### PR TITLE
Refactor openshift_node

### PR DESCRIPTION
**Pull request by mtnbikenc**
- Get pull secret from cluster instead of file
- Use temp_dir var for temp directory
- Remove temp directory
- Upgrade: use (updated) local MCD image

### CODE REVIEW

File: roles/openshift_node/tasks/config.yml [hunk 1]
```diff
@@ -40,15 +41,27 @@
...
+  until:
+  - oc_get.stdout != ''
+  retries: 36
+  delay: 5
+
+- name: Write pull-secret to file
   copy:
-    src: "{{ openshift_pull_secret_path }}"
-    dest: "{{ tempfile.path }}/pull-secret.json"
+    content: "{{ (oc_get.stdout | b64decode }}"
```
Comment (jcpowermac) [inline roles/openshift_node/tasks/config.yml]: Missing closing `)`

File: roles/openshift_node/tasks/config.yml [hunk 2]
```diff
@@ -40,15 +41,27 @@
...
   uri:
     url: "{{ openshift_node_bootstrap_endpoint }}"
-    dest: "{{ tempfile.path }}/bootstrap.ign"
+    dest: "{{ temp_dir.path }}/bootstrap.ign"
     validate_certs: false
 
-- name: Copy pull secret in the directory
+- name: Get cluster pull-secret
+  command: >
+    oc get secret pull-secret
```
Comment (jcpowermac) [inline roles/openshift_node/tasks/config.yml]: I could have missed it but what namespace is this secret in?

Comment (mtnbikenc) [reply roles/openshift_node/tasks/config.yml]: Added namespace

File: roles/openshift_node/tasks/config.yml (hunk 1 above)
Comment (mtnbikenc) [reply roles/openshift_node/tasks/config.yml]: Didn't need the opening `(`

File: README.md [hunk 3]
```diff
@@ -38,7 +38,6 @@ An example inventory can be found in [inventory/hosts.example](inventory/hosts.e
 Required variables include:
 
 - `openshift_kubeconfig_path` - Path to the kubeconfig for the cluster
-- `openshift_pull_secret_path` - Path to the pull secret to the image registry
```
Comment (mtnbikenc) [inline README.md]: @kalexand-rh FYI, `openshift_pull_secret_path` is no longer a required input variable once this merges.

Comment (kalexand-rh) [reply README.md]: https://bugzilla.redhat.com/show_bug.cgi?id=1712579

**Comment by mtnbikenc** [comment]
```
Failing tests:

[k8s.io] EquivalenceCache [Serial] validates pod anti-affinity works properly when new replica pod is scheduled [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-scheduling] SchedulerPreemption [Serial] validates basic preemption works [Suite:openshift/conformance/serial] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
/retest
```
Failing tests:

[sig-autoscaling] [HPA] Horizontal pod autoscaling (scale resource: CPU) [sig-autoscaling] ReplicationController light Should scale from 2 pods to 1 pod [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-network] Service endpoints latency should not be very high  [Conformance] [Suite:openshift/conformance/serial/minimal] [Suite:k8s] [Serial]
[sig-storage] CSI Volumes [Driver: pd.csi.storage.gke.io][Serial] [Testpattern: Dynamic PV (default fs)] subPath should support existing directory [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-storage] CSI Volumes [Driver: pd.csi.storage.gke.io][Serial] [Testpattern: Dynamic PV (default fs)] subPath should support non-existent path [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-storage] CSI Volumes [Driver: pd.csi.storage.gke.io][Serial] [Testpattern: Pre-provisioned PV (xfs)] volumes should be mountable [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: block] [Testpattern: Pre-provisioned PV (default fs)] subPath should support readOnly directory specified in the volumeMount [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: gce-localssd-scsi-fs] [Serial] [Testpattern: Dynamic PV (ext4)] volumes should be mountable [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: gce-localssd-scsi-fs] [Serial] [Testpattern: Inline-volume (ntfs)][sig-windows] volumes should be mountable [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: gce-localssd-scsi-fs] [Serial] [Testpattern: Pre-provisioned PV (default fs)] subPath should support existing directory [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-storage] PersistentVolumes-local  Pods sharing a single local PV [Serial] all pods should be running [Suite:openshift/conformance/serial] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
```
Failing tests:

[sig-api-machinery] Namespaces [Serial] should ensure that all pods are removed when a namespace is deleted [Conformance] [Suite:openshift/conformance/serial/minimal] [Suite:k8s]
[sig-storage] PersistentVolumes-local  Pods sharing a single local PV [Serial] all pods should be running [Suite:openshift/conformance/serial] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
```
Failing tests:

[sig-api-machinery] Namespaces [Serial] should always delete fast (ALL of 100 namespaces in 150 seconds) [Feature:ComprehensiveNamespaceDraining] [Suite:openshift/conformance/serial] [Suite:k8s]
```

**Comment by DanyC97** [comment]
@mtnbikenc very tidy Mister, nice one ! 💯

**Comment by mtnbikenc** [comment]
```
Failing tests:

[sig-api-machinery] Namespaces [Serial] should always delete fast (ALL of 100 namespaces in 150 seconds) [Feature:ComprehensiveNamespaceDraining] [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-autoscaling] [HPA] Horizontal pod autoscaling (scale resource: CPU) [sig-autoscaling] ReplicationController light Should scale from 2 pods to 1 pod [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-cli] Kubectl client [k8s.io] Simple pod should contain last line of the log [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run  [Conformance] [Suite:openshift/conformance/serial/minimal] [Suite:k8s]
[sig-storage] PersistentVolumes-local  Stress with local volumes [Serial] should be able to process many pods and reuse local volumes [Suite:openshift/conformance/serial] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
Investigating tests.
/hold

**Comment by mtnbikenc** [comment]
Waiting for updated 4.2 crio RPM.
/hold

**Comment by mtnbikenc** [comment]
crio rpm is updated
/retest

**Comment by mtnbikenc** [comment]
New crio 1.14.5.
/retest
/hold cancel

**Comment by mtnbikenc** [comment]
```
Failing tests:

[sig-auth] ServiceAccounts should allow opting out of API token automount  [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-autoscaling] [HPA] Horizontal pod autoscaling (scale resource: CPU) [sig-autoscaling] ReplicationController light Should scale from 2 pods to 1 pod [Suite:openshift/conformance/parallel] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
```
Failing tests:

[sig-autoscaling] [HPA] Horizontal pod autoscaling (scale resource: CPU) [sig-autoscaling] ReplicationController light Should scale from 2 pods to 1 pod [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-scheduling] SchedulerPriorities [Serial] Pod should be preferably scheduled to nodes pod can tolerate [Suite:openshift/conformance/serial] [Suite:k8s]
```

**Comment by vrutkovs** [comment]
/hold

Putting it on hold to have a new template applied

**Comment by mtnbikenc** [comment]
/hold cancel
namespace deleted

**Pull request merged**